--- a/arch/mips/lantiq/xway/reset.c
+++ b/arch/mips/lantiq/xway/reset.c
@@ -377,6 +377,13 @@ static int __init mips_reboot_setup(void
 		ltq_rcu_w32(ltq_rcu_r32(RCU_AHB_ENDIAN) | RCU_VR9_BE_AHB1S,
 			    RCU_AHB_ENDIAN);
 
+	/* fix vr9 GPHY reset issue at cold boot */
+        if (of_machine_is_compatible("lantiq,vr9")) {
+                ltq_rcu_w32(0xC0, 0x48);
+                mdelay(200);
+                ltq_rcu_w32(0x0, 0x48);
+        }
+
 	_machine_restart = ltq_machine_restart;
 	_machine_halt = ltq_machine_halt;
 	pm_power_off = ltq_machine_power_off;
