Index: linux-4.19.66/drivers/net/phy/mvsw61xx.c
===================================================================
--- linux-4.19.66.orig/drivers/net/phy/mvsw61xx.c
+++ linux-4.19.66/drivers/net/phy/mvsw61xx.c
@@ -988,6 +988,9 @@ static int mvsw61xx_probe(struct platfor
 				MV_PORTREG(IDENT, 0)) & MV_IDENT_MASK;
 
 	switch(state->model) {
+	case MV_IDENT_VALUE_6161:
+		model_str = MV_IDENT_STR_6161;
+		break;
 	case MV_IDENT_VALUE_6171:
 		model_str = MV_IDENT_STR_6171;
 		break;
@@ -1062,6 +1065,7 @@ mvsw61xx_remove(struct platform_device *
 }
 
 static const struct of_device_id mvsw61xx_match[] = {
+	{ .compatible = "marvell,88e6161" },
 	{ .compatible = "marvell,88e6171" },
 	{ .compatible = "marvell,88e6172" },
 	{ .compatible = "marvell,88e6176" },
Index: linux-4.19.66/drivers/net/phy/mvsw61xx.h
===================================================================
--- linux-4.19.66.orig/drivers/net/phy/mvsw61xx.h
+++ linux-4.19.66/drivers/net/phy/mvsw61xx.h
@@ -224,6 +224,9 @@ enum {
 
 #define MV_IDENT_MASK			0xfff0
 
+#define MV_IDENT_VALUE_6161		0x1610
+#define MV_IDENT_STR_6161		"MV88E6161"
+
 #define MV_IDENT_VALUE_6171		0x1710
 #define MV_IDENT_STR_6171		"MV88E6171"
 
